content/guide/plugins-authoring.md

```diff
@@ -298,7 +298,65 @@ def start(**kwargs):
 
 # Testing Your Plugin
 
-In most cases, the recommendation is to test your plugin's logic using local workflows and only then, run them as part of a Cloudify [deployment]({{page.terminology_link}}#deployment). The [Plugin Template]({{page.template_link}}/blob/3.2/plugin/tests/test_plugin.py) has an example of doing just that.
+In most cases, the recommendation is to test your plugin's logic using local workflows and only then, run them as part of a Cloudify [deployment]({{page.terminology_link}}#deployment). We have supplied you with a nice and tidy
+decorator to do just that. It is provided by the `cloudify-plugins-common.test_utils`, and it's very intuitive to use, But just in case let's look at an example:
+
+{{< gsHighlight  python >}}
+from cloudify.test_utils import workflow_test
+
+@workflow_test(
+                blueprint_path,
+                copy_plugin_yaml,
+                resources_to_copy,
+                temp_dir_prefix,
+                init_args,
+                inputs
+                )
+def test_my_task(self, cfy_local):
+    pass
+{{< /gsHighlight >}}
+
+### Now lets break down the arguments:
+- blueprint_path - A path to the blueprint to run - This is the ONLY mandatory input.
+- copy_plugin_yaml - Tries to locate and copy the plugin.yaml file to the temporary test dir created.
+- resources_to_copy - If any additional resources are needed for the test, they could be supplied here via list.
+- temp_dir_prefix - If you have any special request for the temprary test dir prefix, you should supply it here.
+- init_args - If you have any specific args to be pass to the local.init() method, pass them through here.
+- inputs - A syntatctic sugar for the init_args['inputs'] field.
+
+#### Tips and trick:
+- Passing inputs isn't confined to static ones:
+    - You might want to pass a function name to the inputs arg, this function will be called, and the return value would
+        be set as the inputs for the init. e.g.:
+            {{< gsHighlight  python >}}
+            from cloudify.test_utils import workflow_test
+            def set_inputs():
+                inputs = {}
+                ...
+                return inputs
+            @workflow_test(some_blue_print_path, inputs=set_inputs)
```

Comment:
i dont udnerstand what is gained from this.
why do i need the decorator to be "callable-aware", when i can just pass "inputs=set_inputs()" and evaluate the function before the parameter is passed?
if there is a benefit, it should be stated here
also, rather than a "tip/trick", this is pretty much the signature of the decorator, and should be documented as part of the parameter information